docs/raw/push-notification-server.md

```diff
@@ -0,0 +1,672 @@
+
+### PushNotificationAcknowledgement
+
+`id`: the `ack_required` string passed by the client
+
+## Partitioned topic
+
+This is a modification of the [partitioned topic](./10-waku-usage.md) used by clients
+
+```golang
+var partitionsNum *big.Int = big.NewInt(5000)
+var partition *big.Int = big.NewInt(0).Mod(publicKey.X, partitionsNum)
+
+partitionTopic := "push-notifications-" + strconv.FormatInt(partition.Int64(), 10)
+
+var hash []byte = keccak256(partitionTopic)
+var topicLen int = 4
+
+if len(hash) < topicLen {
+    topicLen = len(hash)
+}
+
+var topic [4]byte
+for i = 0; i < topicLen; i++ {
+    topic[i] = hash[i]
+}
+```
+
+[//]: This section can be removed, for now leaving it here in order to help with the 
+review process. Point can be integrated, suggestion welcome.
+
+## FAQ
+
+### Why having ACL done at the server side and not the client?
+
+We looked into silent notification for 
+[IOS](https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/pushing_background_updates_to_your_app) (android has no equivalent)
+but can't be used as it's expected to receive maximum 2/3 per hour, so not our use case. There
+are also issue when the user force quit the app.
+
+### Why using an access token?
+
+The access token is used to decouple the requesting information from the user from
+actually sending the push notification.
+
+Some ACL is necessary otherwise it would be too easy to spam users (it's still fairly 
+trivial, but with this method you could allow only contacts to send you push notifications).
+
+Therefore your identity must be revealed to the server either when sending or querying.
+
+By using an access token we increase deniability, as the server would know 
+who requested the token but not necessarily who sent a push notification. 
+Correlation between the two can be trivial in some cases.
+
+This also allows a mode of use as we had before, where the server does not propagate
+info at all, and it's left to the user to propagate the token, through contact requests
+for example.
+
+### Why so many different protobuf messages?
+
+Many are just wrappers, I have not re-used any for now for clarity.
+
+
+### Why advertise with the bundle?
+
+Advertising with the bundle allows us to piggy-back on an already implemented behavior
+and save some bandwidth in cases where is not filtering by public keys 
+
+### What's the  bandwidth impact for this?
+
+Generally speaking, for each 1-to-1 message and group chat message you will sending
+1 and `number of participants` push notifications. This can be optimized if 
+multiple users are using the same push notification server. Queries have also 
+a bandwidth impact but they are made only when actually needed
+
+### What's the information disclosed?
+
+The data disclosed with each message sent by the client is above, but for a summary:
+
+When you register with a push notification service you may disclose:
+
+1) Your public key
+2) Which devices you have
```

Comment:
can you choose to disclose only some of those devices? for whatever reason

Reply:
Yes, you can always decide not to register with a notification server, or partially register etc